src/sentry/static/sentry/app/views/organizationGroupDetails/groupDetails.tsx

```diff
@@ -207,37 +241,128 @@ class GroupDetails extends React.Component<Props, State> {
+  renderError() {
+    if (!this.state.error) {
+      return null;
+    }
+
+    switch (this.state.errorType) {
+      case ERROR_TYPES.GROUP_NOT_FOUND:
+        return (
+          <LoadingError message={t('The issue you were looking for was not found.')} />
+        );
+      default:
+        return <LoadingError onRetry={this.remountComponent} />;
+    }
+  }
+
+  getEventView() {
+    const {event} = this.state;
+
+    if (!event) {
+      return undefined;
+    }
+
+    const traceID = event.contexts?.trace?.trace_id;
+
+    if (!traceID) {
+      return undefined;
+    }
+
+    const {organization} = this.props;
+    const orgFeatures = new Set(organization.features);
+    const dateCreated = moment(event.dateCreated).valueOf() / 1000;
+    const pointInTime = event?.dateReceived
+      ? moment(event.dateReceived).valueOf() / 1000
+      : dateCreated;
+
+    const {start, end} = getTraceDateTimeRange({
+      start: pointInTime,
+      end: pointInTime,
+    });
+
+    return EventView.fromSavedQuery({
+      id: undefined,
+      name: `Events with Trace ID ${traceID}`,
+      fields: [
+        'title',
+        'event.type',
+        'project',
+        'project.id',
+        'trace.span',
+        'timestamp',
+        'lastSeen',
+        'issue',
+      ],
+      orderby: '-timestamp',
+      query: `trace:${traceID}`,
+      projects: orgFeatures.has('global-views')
+        ? [ALL_ACCESS_PROJECTS]
+        : [Number(event?.projectID)],
+      version: 2,
+      start,
+      end,
+    });
+  }
+
+  renderInnerContent(
+    project: AvatarProject,
+    relatedEvents: Array<TableDataRow>,
+    eventView?: EventView
+  ) {
     const {children, environments} = this.props;
-    const {group} = this.state;
+    const {group, event} = this.state;
 
     return (
       <React.Fragment>
-        <GroupHeader project={project} group={group} />
+        <GroupHeader
+          project={project}
+          group={group}
+          relatedEventsQuantity={relatedEvents.length}
+        />
         {React.isValidElement(children)
           ? React.cloneElement(children, {
               environments,
               group,
               project,
+              event,
+              relatedEvents,
+              eventView,
             })
           : children}
       </React.Fragment>
     );
   }
 
-  renderError() {
-    if (!this.state.error) {
-      return null;
-    }
+  renderContent(project: AvatarProject) {
+    const eventView = this.getEventView();
 
-    switch (this.state.errorType) {
-      case ERROR_TYPES.GROUP_NOT_FOUND:
-        return (
-          <LoadingError message={t('The issue you were looking for was not found.')} />
-        );
-      default:
-        return <LoadingError onRetry={this.remountComponent} />;
+    if (!eventView) {
+      return this.renderInnerContent(project, []);
     }
+
+    const {location, organization} = this.props;
+    const {event} = this.state;
+
+    return (
+      <DiscoverQuery
```

Comment:
It might be a good idea to check if the org features has `discover-basic` before `<DiscoverQuery/>`

Reply:
sure thing ;-)

Reply:
Maybe `performance-view` instead.

Reply:
After talking to Filip, we decided to make this feature available to all customers, with the exception of the "Open in Discover button" which should only be available if the user has "discover-basic" 😉

Reply:
The discover APIs won't work for users on old plans though. Only some of the plans in prod contain either `discover-basic` and or `performance-view` which are the feature flags that give an organization access to discover.

Reply:
Do we need to do this discover query for all sub tabs of an issue? Does comments or tags views need/use this data? Or could we only do this query on the related events tab?